pkg/storegateway/gateway_test.go

```diff
@@ -361,8 +361,8 @@ func TestStoreGateway_InitialSyncWithWaitRingStability(t *testing.T) {
 				assert.Equal(t, float64(2*testData.numGateways), metrics.GetSumOfGauges("cortex_bucket_stores_tenants_discovered"))
 
 				if testData.shardingStrategy == util.ShardingStrategyShuffle {
-					assert.Equal(t, float64(testData.tenantShardSize*numBlocks), metrics.GetSumOfGauges("cortex_blocks_meta_synced"))
-					assert.Equal(t, float64(testData.tenantShardSize*numUsers), metrics.GetSumOfGauges("cortex_bucket_stores_tenants_synced"))
+					assert.Equal(t, float64(int(testData.tenantShardSize)*numBlocks), metrics.GetSumOfGauges("cortex_blocks_meta_synced"))
+					assert.Equal(t, float64(int(testData.tenantShardSize)*numUsers), metrics.GetSumOfGauges("cortex_bucket_stores_tenants_synced"))
```

Comment:
Feels like you should also add test cases here where tenantShardSize is a percentage to ensure this keeps making sense.